sycl/include/sycl/ext/oneapi/reduction.hpp

```diff
@@ -671,6 +677,18 @@ class reduction_impl_algo<
     return {*CounterBuf, CGH};
   }
 
+  auto getGroupsCounterAccDiscrete(handler &CGH) {
+    auto Buf = std::make_shared<buffer<int, 1>>(1);
+    CGH.addReduction(Buf);
+    std::shared_ptr<detail::queue_impl> QueueCopy = CGH.MQueue;
+    auto Event = CGH.withAuxHandler(QueueCopy, [&](handler &InitHandler) {
+      auto Acc = Buf->get_access<access::mode::discard_write>(InitHandler);
+      InitHandler.single_task([=]() { Acc[0] = 0; });
```

Comment:
This would work because unnamed-lambda functions mode is ON by default now (because the default value of it depends on sycl-std, which is set to 2020 by default now).

It will not work with -fno-sycl-unnamed-lambda, IMO.